resource_detectors/lib/opentelemetry/resource/detectors/google_cloud_platform.rb

```diff
@@ -38,12 +38,29 @@ def detect # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metr
             resource_attributes[OpenTelemetry::SemanticConventions::Resource::CONTAINER_NAME] = ENV['CONTAINER_NAME']
           end
 
+          if gcp_env.knative?
+            resource_attributes[OpenTelemetry::SemanticConventions::Resource::CLOUD_PROVIDER] = 'gcp'
+            resource_attributes[OpenTelemetry::SemanticConventions::Resource::CLOUD_ACCOUNT_ID] = gcp_env.project_id
+            resource_attributes[OpenTelemetry::SemanticConventions::Resource::FAAS_NAME] = gcp_env.knative_service_id
+            resource_attributes[OpenTelemetry::SemanticConventions::Resource::FAAS_VERSION] = gcp_env.knative_service_revision
+            zone = gcp_env.instance_zone
+            resource_attributes[OpenTelemetry::SemanticConventions::Resource::CLOUD_REGION] = get_region zone
+            resource_attributes[OpenTelemetry::SemanticConventions::Resource::CLOUD_AVAILABILITY_ZONE] = zone
```

Comment:
Some issue around possibly setting this as a nil value.

Reply:
Similarly, if zone is nil then it won't get set. 
`resource_attributes.delete_if { |_key, value| value.nil? || value.empty? }`